### ISSUE TITLE
Spread with in constructors bombs.

### ISSUE THREAD

**Issue by hemanth** [6to5/6to5#373]
```js
let date = new Date(...[2015, 1, 1]);
```

Throws an Error:

```
TypeError: this is not a Date object.
    at Date.valueOf (native)
    at Date.toJSON (native)
    at Object.stringify (native)
    at Console.console.log (https://6to5.org/repl.html:193:27)
    at eval (eval at <anonymous> (https://6to5.org/repl.html:199:33), <anonymous>:13:9)
    at compile (https://6to5.org/repl.html:199:39)
```
I have enabled the experimental flag.

**Comment by zloirock** [comment]
https://github.com/6to5/6to5/pull/203

**Comment by sebmck** [comment]
Weird, I wonder if it's because it's a native type.

**Comment by hemanth** [comment]
Did [this](https://github.com/6to5/6to5/pull/203#issuecomment-64079487) go in?

**Comment by sebmck** [comment]
@hemanth Sure did. I'll investigate shortly, it's likely due to it being a native type.

**Comment by zloirock** [comment]
@sebmck Yes, I wrote about it. Native constructors create object from prototype and set hidden properties. `Object.create` can't set them. The current implementation correctly works with native constructors only if those constructor works with `new` as without (for example, `Array`). Be better add it to caveats.

**Comment by sebmck** [comment]
Probably just going to turn:

```javascript
new Date(...[2015, 1, 1]);
```

into

```javascript
var _bind = Function.prototype.bind;
new (_bind.apply(Date, [null, 2011, 4, 3]))();
```

**Comment by thejameskyle** [comment]
I'll take this.

**Comment by sebmck** [comment]
@thejameskyle Already fixed. My suggestion is done only on native types for performance etc.

**Comment by thejameskyle** [comment]
Damn it, I was so close to the same thing. You have a commit?

**Comment by hemanth** [comment]
:+1:

**Comment by sebmck** [comment]
Thanks! Fixed as of 2.6.0
